### DISCUSSION TITLE
NodeJS: build inventory application instead of coding along

### DISCUSSION THREAD

**Discussion by OmarNaguib** [TheOdinProject/curriculum#26292] [General]
### Describe your suggestion

In the express & mongoose section, we have 2 projects, local-library and inventory application.

Both of them are mostly the same, require the same sort of tasks and are done the same way in the samer order.

My understanding is that TOP avoids depending on tutorial for making projects, and for good reasons that were touched upon in foundations.

Instead TOP's approach is to to learn relevant and necessary information, then make something that works.

my suggestion is redesigning the section so that students make the inventory application gradually as they go through the express tutorial, and that coding along with the tutorial would be optional.

Another way to do it would be to make coding along optional but leave things as they are with "Inventory application" placed at the end.

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/paths/full-stack-javascript/courses/nodejs#express-mongoose

### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### (Optional) Discord Name

Omar (onsmak) @onsmak

**Comment by rlmoser99** [comment]
@TheOdinProject/javascript Can someone take a look please

**Comment by 01zulfi** [comment]
Removing the code along in that section in favor of in-house content has been on our radar for a while. 

@OmarNaguib Thanks for the suggestion. This change will be likely made within a larger Node course refresh. The team still needs time to figure out what that would look like. Until then, I'm marking this issue as discussion.

**Reply by OmarNaguib** [reply]
@01zulfi
thanks for the clarification.

I have 2 questions

When this refresh is okayed, will it be available for people to contribute, or just the team?
If so how would we know about

Also i suppose there is no need to make an issue about the mongodb course as it would be a part of that refresh?

**Reply by 01zulfi** [reply]
> When this refresh is okayed, will it be available for people to contribute, or just the team?
> If so how would we know about

That depends on a few factors: the scope of the refresh, maintainers' bandwidth. I can't say for certain though, but we normally do open up for contributions in a certain capacity. GitHub (sometimes Discord) is the best bet on knowing when that happens.

> Also i suppose there is no need to make an issue about the mongodb course as it would be a part of that refresh?

Not sure what you mean here. Can you elaborate?

**Reply by OmarNaguib** [reply]
> 
> That depends on a few factors: the scope of the refresh, maintainers' bandwidth. I can't say for certain though, but we normally do open up for contributions in a certain capacity. GitHub (sometimes Discord) is the best bet on knowing when that happens.
> 
So that means there maintainers would make issues with the "help wanted" tag then right?
> 
> Not sure what you mean here. Can you elaborate?
>
currently the curriculum teaches mongoDB using outside content, I had some reservations against that mongoDB course and was meaning to make an issue about it, so should I or not?

**Reply by 01zulfi** [reply]
> So that means there maintainers would make issues with the "help wanted" tag then right?

Most likely. Historically, we've also used [top-meta repo](https://github.com/TheOdinProject/top-meta/) for issues

--- 

Please do make an issue on MongoDB.

**Reply by OmarNaguib** [reply]
Ok thank you for your time and all the information.
I'll make that issue
